### PR TITLE
HHH-18041 With SharedCacheMode.DISABLE_SELECTIVE entities with cacheable false should not be cached

### PR DESCRIPTION
**Pull request by dreab8**
https://hibernate.atlassian.net/browse/HHH-18041

### CODE REVIEW

File: hibernate-core/src/main/resources/org/hibernate/hibernate-mapping-4.0.xsd [hunk 1]
```diff
@@ -1396,7 +1396,7 @@ arbitrary number of queries, and import declarations of arbitrary classes.
     </xs:sequence>
     <xs:attribute name="cache-mode" type="cache-mode-attribute"/>
     <xs:attribute name="cache-region" type="xs:string"/>
-    <xs:attribute name="cacheable" default="false" type="xs:boolean"/>
```
Comment (beikov) [inline hibernate-core/src/main/resources/org/hibernate/hibernate-mapping-4.0.xsd:1399]: Is this change really necessary?

Comment (dreab8) [reply hibernate-core/src/main/resources/org/hibernate/hibernate-mapping-4.0.xsd:1399]: it seems so

Comment (dreab8) [reply hibernate-core/src/main/resources/org/hibernate/hibernate-mapping-4.0.xsd:1399]: without we have test failures

Comment (beikov) [reply hibernate-core/src/main/resources/org/hibernate/hibernate-mapping-4.0.xsd:1399]: Because this change is actually a schema change for hbm.xml which we shouldn't do lightly.

**Review by beikov — APPROVED** [review]

**Comment by sebersole** [comment]
+1
